feat(utxo-core): implement buildToSignPsbt for BIP322 message signing

davidkaplanbitgo opened the pull request:
Add functionality to construct the toSign PSBT for BIP322 message verification with support for redeem scripts and witness scripts. Includes tests to verify correctness against BIP322 test vectors.

Supports v0 segwit and non-segwit addresses. Taproot to be added later.

Issue: BTC-2362

<!--
# Please be aware of the following when making your pull request:

## Description

Please include a summary of your proposed changes and which issue is being addressed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Issue Number

Internal Users - Please include the related internal tracking number (e.g. BG-000000).
External Users - Please link to any relevant github issues as necessary.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My code compiles correctly for both Node and Browser environments
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [ ] The ticket or github issue was included in the commit message as a reference
- [ ] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
-->

davidkaplanbitgo committed:
feat(utxo-core): implement buildToSignPsbt for BIP322 message signing

Add functionality to construct the toSign PSBT for BIP322 message verification
with support for redeem scripts and witness scripts. Includes tests to verify
correctness against BIP322 test vectors.

Supports v0 segwit and non-segwit addresses. Taproot to be added later.

Issue: BTC-2360

Co-authored-by: llm-git <llm-git@ttll.de>

OttoAllmendinger commented on modules/utxo-core/src/bip322/toSign.ts:
```diff
@@ -0,0 +1,49 @@
+ * @param {AddressDetails} addressDetails - The details of the address, including redeemScript and/or witnessScript.
+ * @returns {string} - The hex representation of the constructed PSBT.
+ */
+export function buildToSignPsbt(toSpendTxHex: string, addressDetails: AddressDetails): string {
```
stringly typed:

pass a higher level type like `Transaction` instead and return a higher level type like `Psbt`

davidkaplanbitgo committed:
feat(utxo-core): improve bip322 transaction handling

Change buildToSignPsbt to directly accept a Transaction object instead of hex
string, and return a Psbt object instead of hex. Similarly update
buildToSpendTransaction to return a Transaction object.

Issue: BTC-2362

Co-authored-by: llm-git <llm-git@ttll.de>
